packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -0,0 +1,297 @@
+import MemoryStore from './MemoryStore'
+import Store from './Store'
+import queue from 'src/decorators/queue'
+import rateLimitRetry from 'src/decorators/rateLimitRetry'
+import wait from 'wait'
+import { BigNumber, Signer, providers } from 'ethers'
+import { EventEmitter } from 'events'
+import { getBumpedGasPrice, getProviderChainSlug } from 'src/utils'
+import { v4 as uuidv4 } from 'uuid'
+
+export enum State {
+  Confirmed = 'confirmed',
+  Boosted = 'boosted'
+}
+
+type InflightItem = {
+  hash?: string
+  sentAt: number
+  confirmed: boolean
+}
+
+type MarshalledItem = {
+  id: string
+  createdAt: number
+  txHash: string
+  from: string
+  to: string
+  data: string
+  value: string
+  nonce: number
+  gasPrice: string
+  gasLimit: string
+}
+
+class GasBoostTransaction extends EventEmitter implements providers.TransactionResponse {
+  started: boolean = false
+  pollMs: number = 10 * 1000
+  timeTilBoostMs: number = 1 * 60 * 1000
+  gasPriceMultiplier: number = 1.5
+  maxGasPriceGwei: BigNumber = BigNumber.from(500)
+  boostIndex: number = 0
+  inflightItems: InflightItem[] = []
+  signer: Signer
+  store: Store = new MemoryStore()
+  chainSlug: string
+  id: string
+  createdAt: number
+  txHash: string
+  from: string
+  to: string
+  data: string
+  value: BigNumber
+  nonce: number
+  gasLimit: BigNumber
+  gasPrice: BigNumber
+  chainId: number
+  confirmations: number
+
+  constructor (tx: providers.TransactionRequest, signer: Signer, store?: Store) {
+    super()
+    this.signer = signer
+    if (store) {
+      this.store = store
+    }
+    this.id = uuidv4()
+    this.createdAt = Date.now()
+    this.from = tx.from
+    this.to = tx.to
+    if (tx.data) {
+      this.data = tx.data.toString()
+    }
+    if (tx.value) {
+      this.value = BigNumber.from(tx.value.toString())
+    }
+    if (tx.nonce) {
+      this.nonce = Number(tx.nonce.toString())
+    }
+    if (tx.gasPrice) {
+      this.gasPrice = BigNumber.from(tx.gasPrice.toString())
+    }
+    if (tx.gasLimit) {
+      this.gasLimit = BigNumber.from(tx.gasLimit.toString())
+    }
+  }
+
+  get hash ():string {
+    if (!this.txHash) {
```

Comment:
I believe this should be `if (this.txHash)`

Reply:
nice catch, thanks